src/wifi.rs

```diff
@@ -742,6 +742,15 @@ impl<'d> WifiDriver<'d> {
         Ok(result)
     }
 
+    fn set_sta_conf_if_changed(&mut self, conf: &ClientConfiguration) -> Result<(), EspError> {
```

Comment:
Let's make this even simpler and just move the equality check inside the existing `set_sta_conf` / `set_ap_conf` methods?